nix: Add postgrest-git-hooks to enable pre-commit and pre-push hooks

wolfgangwalther opened the pull request:
As promised in https://github.com/PostgREST/postgrest/pull/1822#discussion_r616165264 here's `postgrest-git-hooks`. Run `postgrest-git-hooks enable` in nix-shell to enable basic pre-commit and pre-push hooks.

Careful: When testing this, make sure to run `postgrest-git-hooks disable` before starting to work on another branch again. When the other branch doesn't have `postgrest-git-hooks`, this will prevent any commit or push going through. Just in case somebody is getting stuck somewhere: This can be overriden with `--no-verify` all the time.

Related: #1804

monacoremo commented:
Works great for me!

Nitpicks, does not block merging from my view:

On first use of `postgrest-git-hooks enable basic` I got 

```
sed: can't read .git/hooks/pre-commit: No such file or directory
sed: can't read .git/hooks/pre-push: No such file or directory
```

The bash completion does not seem to be working, it's proposing files for me instead of enable, disable etc.

wolfgangwalther commented:
> On first use of `postgrest-git-hooks enable basic` I got
> 
> ```
> sed: can't read .git/hooks/pre-commit: No such file or directory
> sed: can't read .git/hooks/pre-push: No such file or directory
> ```
Fixed.

> The bash completion does not seem to be working, it's proposing files for me instead of enable, disable etc.

Fixed.
